tests/apollo/util/bft.py

```diff
@@ -1321,4 +1321,29 @@ async def the_replica_to_ask_for_view_change():
                                     replica_asks_to_leave_view_msg_count=replica_asks_to_leave_view_msg_count)
                                 return replica_asks_to_leave_view_msg_count
 
-        return await self.wait_for(the_replica_to_ask_for_view_change, 60, .5)
\ No newline at end of file
+        return await self.wait_for(the_replica_to_ask_for_view_change, 60, .5)
+
+    async def wait_for_replicas_to_reach_view(self, replicas_ids, expected_view):
+        """
+        Wait for all replicas to reach the expected view.
+        """
+        with log.start_action(action_type="wait_for_replicas_to_reach_view", replicas_ids=replicas_ids, expected_view=expected_view) as action:
+            with trio.fail_after(seconds=30):
+                async with trio.open_nursery() as nursery:
+                    for replica_id in replicas_ids:
+                        nursery.start_soon(self.wait_for_replica_to_reach_view, replica_id, expected_view)
+            
+    async def wait_for_replica_to_reach_view(self, replica_id, expected_view):
+        """
+        Wait for a single replica to reach the expected view.
+        """
+        with log.start_action(action_type="wait_for_replica_to_reach_view", replica=replica_id, expected_view=expected_view) as action:
+            async def expected_view_to_be_reached():
+                key = ['replica', 'Gauges', 'view']        
```

Comment:
Are we sure this metric is incremented when the replica does `MoveToHigherView`?

Reply:
Yes, this is the correct metric. The following LOC (Currently 2431) in `MoveToHigherView`'s body sets the value of the metric:
`metric_view_.Get().Set(nextView);`